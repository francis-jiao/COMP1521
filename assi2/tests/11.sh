# inserts/deletes into a small tree
./test4 < tree2
