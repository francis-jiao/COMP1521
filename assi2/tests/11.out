13 11 10 12 15 14 16 
#nodes = 7
Original Tree:
       13
       / \
      /   \
     /     \
    /       \
   11       15
  / \       / \
 /   \     /   \
10   12   14   16

> i 18
New Tree:  #nodes=8,    depth=4
       13
       / \
      /   \
     /     \
    /       \
   11       15
  / \       / \
 /   \     /   \
10   12   14   16
                 \
                 18

> d 15
New Tree:  #nodes=7,    depth=3
       13
       / \
      /   \
     /     \
    /       \
   11       16
  / \       / \
 /   \     /   \
10   12   14   18

> d 13
New Tree:  #nodes=6,    depth=3
     14
     / \
    /   \
   11   16
  / \     \
 /   \    18
10   12

> i 15
New Tree:  #nodes=7,    depth=3
       14
       / \
      /   \
     /     \
    /       \
   11       16
  / \       / \
 /   \     /   \
10   12   15   18

> q
